src/mergecil.ml

```diff
@@ -1507,46 +1453,30 @@
                if debugInlines then ignore (E.log " Not found\n");
                H.add inlineBodies printout inode;
                mergePushGlobal g')
-           else if
-             (* either the function is not inline, or we're not attempting to
-              * merge inlines *)
-             mergeGlobals && (not fdec'.svar.vinline)
-             && fdec'.svar.vstorage <> Static
-           then (
-             (* sm: this is a non-inline, non-static function.  I want to
-                * consider dropping it if a same-named function has already
-                * been put into the merged file *)
-             let curSum = functionChecksum fdec' in
-             (*(trace "mergeGlob" (P.dprintf "I see extern function %s, sum is %d\n"*)
-             (*                              fdec'.svar.vname curSum));*)
-             try
-               let prevFun, prevLoc, prevSum =
-                 H.find emittedFunDefn fdec'.svar.vname
-               in
+           else if mergeGlobals && (not fdec'.svar.vinline) && fdec'.svar.vstorage <> Static then (
```

Comment:
I think it might be better to De Morgan this condition such that it's more obvious that it matches the one in `processVarinfo`:
```suggestion
           else if mergeGlobals && not (fdec'.svar.vstorage = Static || fdec'.svar.vinline) then ( (* !merge_inlines is false here anyway *)
```